src/scripts/gen-react-bindings.js

```diff
@@ -8,6 +9,40 @@ fs.mkdir(REACT_BINDINGS_DIRECTORY, { recursive: true })
 const COMPONENT_PREFIX = 'leo'
 const SVELTE_REACT_WRAPPER_PATH = '../shared/svelte-react.js'
 
+const getEventsTypeDefinition = (componentName, componentEventNames) => {
+  if (componentEventNames.length === 0) return null
+
+  return `
+export type ${componentName}EventProps = {
+${componentEventNames
+  .map((event) => `  ${event}?: (event: CustomEvent) => void`)
+  .join(';\n')}
+}
+`
+}
+
+const findEventsTypeDefinition = async (svelteFilePath, componentName) => {
+  const relativePath = path.relative('../components', svelteFilePath)
+  const fileContents = await fs.readFile(relativePath)
+
+  const eventDispatcherRegex = /createEventDispatcher<{((\n.*)+)\s}>/g
+  const findDefinedEvents = fileContents
+    .toString()
+    .match(eventDispatcherRegex)?.[0]
+
+  if (findDefinedEvents) {
+    const eventNameRegex = /(\w+)\:.*/g
+
+    const componentEventNames = [
```

Comment:
We can actually type the event as `CustomEvent<DetailType>` too

Reply:
Done, to correctly define the type I rewrote this function to do a regex on the `/types/*.svelte.d.ts` file.

I had to rewrite it this way because in some svelte files the `DetailType` was defined separately, like in [menu.svelte](https://github.com/brave/leo/blob/298a59f8f111a0abbc2fc9b22885ca59e17d0b15/src/components/menu/menu.svelte#L28) but the svelte type file has the event detail types without the separate definition.